components/livekit/src/component.ts

```diff
@@ -0,0 +1,330 @@
+import { ILoggerComponent } from '@well-known-components/interfaces'
+import {
+  AccessToken,
+  IngressClient,
+  IngressInfo,
+  IngressInput,
+  ParticipantInfo,
+  Room,
+  RoomServiceClient,
+  WebhookReceiver
+} from 'livekit-server-sdk'
+import { isErrorWithMessage } from '@dcl/core-commons'
+
+import {
+  CreateIngressOptions,
+  CreateRoomOptions,
+  GenerateCredentialsOptions,
+  ILivekitComponent,
+  LivekitComponentOptions,
+  LivekitCredentials,
+  UpdateParticipantOptions
+} from './types'
+import { LivekitIngressNotFoundError, LivekitWebhookVerificationError } from './errors'
+
+/**
+ * Creates a LiveKit component for managing rooms, participants, and credentials
+ *
+ * @example
+ * ```typescript
+ * const livekit = await createLivekitComponent(
+ *   { logs },
+ *   {
+ *     settings: {
+ *       host: 'wss://livekit.example.com',
+ *       apiKey: 'your-api-key',
+ *       secret: 'your-api-secret'
+ *     }
+ *   }
+ * )
+ *
+ * // Generate credentials for a participant
+ * const credentials = await livekit.generateCredentials({
+ *   identity: 'user-123',
+ *   roomName: 'my-room',
+ *   permissions: { canPublish: true, canSubscribe: true }
+ * })
+ * ```
+ */
+export async function createLivekitComponent(
+  components: { logs: ILoggerComponent },
+  options: LivekitComponentOptions
+): Promise<ILivekitComponent> {
+  const { logs } = components
+  const { settings } = options
```

Comment:
Should we use the config component or pass the credentials and host properties as options?